geth/api/connection.go

```diff
@@ -0,0 +1,65 @@
+package api
+
+import (
+	"fmt"
+)
+
+// ConnectionState represents device connection state and type,
+// as reported by mobile framework.
+//
+// Zero value represents default assumption about network
+// connection until first update — online, on cellular and not expensive.
```

Comment:
Why do we assume that `ConnectionCellular` is a default connection?

Reply:
Hmm, where do you see it as default?

Reply:
`Zero value represents default assumption about network`. 

`Type` field will have value qual to `0` which is `ConnectionCellular` because of `ConnectionCellular ConnectionType = iota`.

Reply:
Ah, indirect, yes. That's why I often start with `iota + 1`.

Reply:
Yeah or set `iota` as some undefined state if applicable.

Reply:
@adambabik that's a good question. For mobile, I guess it's obvious that mobile phone has cellular network connection more often than wifi, so it should have a higher probability.
The question though is what about `statusd` as a standalone node (where ConnectionChange will be never called because there is no status-react part). We probably would want to differentiate mobile vs standalone node run. Does it make sense?

Reply:
Both states, cellular and wifi, document a very low layer of transportation, like also wire. So let's start with the question how this information is helpful for us, how will it influence our runtime behaviour? Also I sometimes got higher transfer rates via 4G than via a bad wifi. So just as a suggestion maybe a kind of throughput/bandwidth information could be more helpful.

Reply:
@themue I think this question should be researched and answered in a follow-up issues/PRs. This PR only adds an interface to handle network state changes, and it's limited by what React-Native can provide.
So this PR doesn't change runtime behavior, it just opens a way to to do it.

Reply:
@divan Good idea. Only found it interesting due to classification of communication type. Also your connection type allows more differantion in future. :+1:

Reply:
Just to make sure we're on the same page – this classification comes from RN: https://facebook.github.io/react-native/docs/netinfo.html

Reply:
Interesting link, thank you for providing. Maybe interesting to add it to a comment too, but not needed.

Reply:
> For mobile, I guess it's obvious that mobile phone has cellular network connection more often than wifi, so it should have a higher probability.

That sounds reasonable but I wonder if we shouldn't assume `unknown` state first and do the transition to `wifi` or `cellular` after launching the app, whenever the network state changes and whenever the app goes to the foreground.

Reply:
> Interesting link, thank you for providing. Maybe interesting to add it to a comment too, but not needed.
@themue this link is actually already there, in two another comments.

@adambabik fair enough. Done.